strawberry/experimental/pydantic/error_type.py

```diff
@@ -44,26 +49,57 @@ def field_type_to_type(type_):
 def error_type(
     model: Type[BaseModel],
     *,
-    fields: List[str],
+    fields: List[str] = None,
     name: Optional[str] = None,
     description: Optional[str] = None,
     federation: Optional[FederationTypeParams] = None,
+    all_fields: bool = False
 ):
     def wrap(cls):
         model_fields = model.__fields__
-        fields_set = set(fields)
+        fields_set = set(fields) if fields else set([])
 
-        cls = dataclasses.make_dataclass(
-            cls.__name__,
-            [
+        existing_fields = getattr(cls, "__annotations__", {})
+        fields_set = fields_set.union(
+            set(name for name, typ in existing_fields.items() if typ == auto)
+        )
+
+        if all_fields:
```

Comment:
Can we add a warning here if `fields_set` is not empty? We'll be overriding any explicitly defined fields and that is probably a bug in the application code.